bot/src/discord_commands/schedule.js

```diff
@@ -0,0 +1,165 @@
+const { Permissions } = require('monochrome-bot');
+const randomHelper = require('./random.js');
+
+let intervals = [];
+
+const NUMBER_OF_RETRIES = 50;
+
+function randomAction(msg, suffix, monochrome) {
+  const suffixLowerCase = suffix.toLowerCase();
+  monochrome.updateUserFromREST(msg.author.id).catch(() => {});
+
+  return randomHelper.getRandomWordRecursive(
+    suffixLowerCase,
+    msg,
+    NUMBER_OF_RETRIES,
+    monochrome,
+  );
+}
+
+function setTimer(id, msg, level, monochrome, freq, startTime) {
+  let nextTime = 0;
+  if (startTime) {
+    nextTime = startTime.getTime() - (new Date()).getTime();
+    while (nextTime < 0) {
+      nextTime += freq;
+    }
+  } else {
+    nextTime = freq;
+  }
+  let index = intervals.findIndex(i => i.id === id);
+  if (index !== -1) {
+    intervals[index].interval = setTimeout(() => {
+      randomAction(msg, level, monochrome);
+      setTimer(id, msg, level, monochrome, freq);
+    }, nextTime)
+  }
+}
+
+async function setSchedule(suffix, msg, monochrome) {
+  let freq = '';
+  let freqType = '';
+  let start = '';
+  let level = '';
+  let suffixArray = suffix.split(" ");
+  if (suffixArray.length === 1 && suffixArray[0] === 'clear') {
+    intervals.forEach(i => clearTimeout(i.interval));
+    intervals = [];
+    msg.channel.createMessage('All scheduled commands have been cleared.');
```

Comment:
This should either be awaited or returned, so if there is an error Kotoba will catch it and handle it. Anywhere where `createMessage()` is used should be awaited or returned.